pytorch_lightning/loggers/base.py

```diff
@@ -23,52 +26,104 @@ def __init__(self):
         self._rank = 0
 
     @property
-    def experiment(self):
+    def experiment(self) -> Any:
         raise NotImplementedError()
 
-    def log_metrics(self, metrics, step):
+    def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None):
         """Record metrics.
 
-        :param float metric: Dictionary with metric names as keys and measured quanties as values
-        :param int|None step: Step number at which the metrics should be recorded
+        Args:
+            metrics: Dictionary with metric names as keys and measured quantities as values
+            step: Step number at which the metrics should be recorded
         """
         raise NotImplementedError()
 
-    def log_hyperparams(self, params):
+    def log_hyperparams(self, params: argparse.Namespace):
         """Record hyperparameters.
 
-        :param params: argparse.Namespace containing the hyperparameters
+        Args:
+            params: argparse.Namespace containing the hyperparameters
         """
         raise NotImplementedError()
 
     def save(self):
         """Save log data."""
 
-    def finalize(self, status):
+    def finalize(self, status: str):
         """Do any processing that is necessary to finalize an experiment.
 
-        :param status: Status that the experiment finished with (e.g. success, failed, aborted)
+        Args:
+            status: Status that the experiment finished with (e.g. success, failed, aborted)
         """
 
     def close(self):
         """Do any cleanup that is necessary to close an experiment."""
 
     @property
-    def rank(self):
+    def rank(self) -> int:
         """Process rank. In general, metrics should only be logged by the process with rank 0."""
         return self._rank
 
     @rank.setter
-    def rank(self, value):
+    def rank(self, value: int):
         """Set the process rank."""
         self._rank = value
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the experiment name."""
         raise NotImplementedError("Sub-classes must provide a name property")
 
     @property
-    def version(self):
+    def version(self) -> Union[int, str]:
```

Comment:
also rather `@abstractmethod` so it does not allow to create the class, better than running for while and then crash...

Reply:
Agreed, done :)